### ISSUE TITLE
different behavior when dismissing the keyboard

### ISSUE THREAD

**Issue by felixjichao** [MessageKit/MessageKit#712]
![plus](https://user-images.githubusercontent.com/17485165/40892956-b10298c4-6752-11e8-9425-87f2f01bf46f.gif)
![x](https://user-images.githubusercontent.com/17485165/40892958-b609c572-6752-11e8-80f4-e30bc3e34efd.gif)

When dismissing the keyboard by dragging the keyboard and releasing it, after the keyboard disappeared, on iPhone X, the collectionView adjusted the inset and showed the last message (the inset is at the bottom); however, on iPhone 8 Plus, the collectionView didn't, but kept the inset when releasing the keyboard.

update:
On handleKeyboardDidChangeState,
``` 
if (keyboardEndFrame.origin.y + keyboardEndFrame.size.height) > UIScreen.main.bounds.height {
    // Hardware keyboard is found
     ... // when releasing the keyboard, on iPhone X, this path is executed
} else {
     //Software keyboard is found
    ... // when releasing the keyboard, on iPhone 8 Plus, this path is executed
}
```

**Comment by SD10** [comment]
@felixjichao What MessageKit version are you using?

**Comment by felixjichao** [comment]
@SD10   I am using 1.0.0-beta.1

**Comment by zhongwuzw** [comment]
@SD10 , #707 make these behavior consistently, but it would still exists unexpected behavior that adjust `contentOffset` when `maintainPositionOnKeyboardFrameChanged` sets to `true`. I can have a trick to fix it by `UIKeyboardWillHide` and `UIKeyboardWillShow`.

**Comment by SD10** [comment]
@zhongwuzw In my opinion, they should both behave like the iPhone X. I think this is a bug 🤔 
I was planning on writing a keyboard manager library for the bottom inset soon. Something like the [KeyboardTracker at Chatto](https://github.com/badoo/Chatto/blob/master/Chatto/Source/ChatController/Collaborators/KeyboardTracker.swift)

**Comment by hyouuu** [comment]
On my iPhone X, on first view appear the contents are all visible above the input bar, but on 2nd appear, they are hidden behind it (seems even below the bottom of the screen by the size it's hidden). I've tried to set `maintainPositionOnKeyboardFrameChanged` to true and false, and even tried to remove the `if isFirstLayout {` check in MessagesViewController, but the result is the same - thoughts for a fix for the time being?

**Comment by martinpucik** [comment]
This should be fixed on 4.0.0 branch
